# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../musl/pkg/man/man-gencat-include-fcntl.patch
# Copyright (C) 2014 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

Include fcntl.h for getting the 'oflags' defined

---------------------------------------------------------------------------
gencat.c: In function 'main':
gencat.c:144:32: error: 'O_WRONLY' undeclared (first use in this function)
gencat.c:144:32: note: each undeclared identifier is reported only once for each function it appears in
gencat.c:144:41: error: 'O_TRUNC' undeclared (first use in this function)
gencat.c:144:49: error: 'O_CREAT' undeclared (first use in this function)
gencat.c:148:35: error: 'O_RDONLY' undeclared (first use in this function)
gencat.c: In function 'writeIfChanged':
gencat.c:201:24: error: 'O_WRONLY' undeclared (first use in this function)
gencat.c:201:33: error: 'O_CREAT' undeclared (first use in this function)
gencat.c:212:30: error: 'O_RDWR' undeclared (first use in this function)
gencat.c:222:27: error: 'O_RDONLY' undeclared (first use in this function)
gencat.c:253:33: error: 'O_TRUNC' undeclared (first use in this function)
make[1]: *** [gencat.o] Error 1
---------------------------------------------------------------------------

--- a/gencat/gencat.c
+++ b/gencat/gencat.c
@@ -52,6 +52,7 @@
 #ifdef __linux__
 #include <string.h>
 #include <unistd.h>
+#include <fcntl.h>
 #endif
 #include <sys/file.h>
 #include <sys/stat.h>
